stubs/Flask-SQLAlchemy/flask_sqlalchemy/model.pyi

```diff
@@ -10,14 +14,14 @@ camelcase_re: Pattern[str]
 def camel_to_snake_case(name: str) -> str: ...
 
 class NameMetaMixin(type):
-    def __init__(cls, name, bases, d) -> None: ...
-    def __table_cls__(cls, *args, **kwargs): ...
+    def __init__(cls, name: str, bases: tuple[type, ...], d: dict[str, Any]) -> None: ...
+    def __table_cls__(cls, *args: Any, **kwargs: Any) -> Table | None: ...
```

Comment:
This doesn't look right to me; I don't think this method can _really_ be called with _any_ arbitrary `*args, **kwargs` combination. The "true" signature looks like it might be quite complicated; I'd rather we just left it unannotated for now, as an explicit marker that there's still stuff to do here in future PRs:

```suggestion
    def __table_cls__(cls, *args, **kwargs) -> Table | None: ...
```

Reply:
That's fair. This was before I had seen in `requests` that the group will sometimes use `*` to name out upstream `*args` and `**kwargs` usage. 

Table Class is logically a good candidate for `ParamSpec` as it's described here:
https://docs.sqlalchemy.org/en/14/orm/declarative_config.html#table-cls

But I'm not sure if the current ParamSpec description allows for capturing a class's init as a parameter spec or not

Reply:
We can leave that to another PR :)